model.py

```diff
@@ -9575,14 +9596,9 @@ def explain(self, include_password=False):
             lines.append('Used by library: "%s"' % library.short_name)
         if self.external_account_id:
             lines.append('External account ID: "%s"' % self.external_account_id)
-        if integration.url:
-            lines.append('URL: "%s"' % integration.url)
-        if integration.username:
-            lines.append('Username: "%s"' % integration.username)
-        if integration.password and include_password:
-            lines.append('Password: "%s"' % integration.password)
         for setting in integration.settings:
-            lines.append('Setting "%s": "%s"' % (setting.key, setting.value))
+            if (setting.key != ExternalIntegration.PASSWORD) or include_password:
+                lines.append('Setting "%s": "%s"' % (setting.key, setting.value))
```

Comment:
Among other ideas I've been having for improving discoverability of these configuration settings, I had the idea of marking certain items as "secret". So ExternalIntegration.PASSWORD would be an object that looks like a normal string, except hasattr(ExternalIntegration.PASSWORD, 'secret') would be True.

This would eliminate the need to put special cases in this code for knowing whether a field value is secret or not.

Not a high priority IMO.